web_client/src/textIntervalReader.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ *  A higher order component that reads text from `readTextInterval`: an async function that returns
+ *  `{ text, readingEvery }`, where `text` is a text string read and `readingEvery` is the number
+ *  of seconds after which new `text` should be read. Every `readingEvery` seconds, this component
+ *  will read text from `readTextInterval` and pass the read text to `WrappedComponent`s `text` prop.
+ */
+const withTextIntervalReader = (WrappedComponent, readTextInterval) => (
```

Comment:
@evanhammer let me know if you have any feedback on this. three things i'd like feedback on

1. Naming (i.e. `onTextIntervalRead`) and signature (decomposing the promise's output rather than taking in specified parameters). Would you change either?
2. Not many of the helper functions here are really testable, since the main 'I/O' is setting the component state, and there's almost no business logic (besides setting up an interval `onFirstTextIntervalRead`.
3. I get this warning when running `yarn test`
```
  console.error node_modules/fbjs/lib/warning.js:33
    Warning: Can't call setState (or forceUpdate) on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
        in Unknown (at App.js:17)
        in div (at MechanicalTheatre.jsx:21)
        in div (at MechanicalTheatre.jsx:12)e-
        in MechanicalTheatre (at App.js:16)
        in div (at App.js:15)
        in App (at App.test.js:7)
```
A stackoverflow thread mainly concluded on: "this is why you keep async functions outside of the component", but I don't know how to do that while setting state. Any ideas?

Reply:
Re: 3, it seems to me that I _am_ essentially cancelling all async operations on `componentWillUnmount` when I clear the interval. Maybe the linter isn't picking up on this?

Reply:
RE memory. I think what is happening is when the component unmounts, the intervals is cleared, but if the request to the server has already fired then when it returns you're calling setState on an unmounted component.

Options:
* Store the state outside of this component. Perhaps in App.
* Check to make sure the component is still mounted before calling setState.
* Look up some other option on the Internet.

Reply:
re: memory, since it's a no op if setState is fired and a rare one at that, i think i'll leave it alone for now